libpromises/var_expressions.c

```diff
@@ -95,6 +96,13 @@ static size_t VarRefHash(const VarRef *ref)
     return (h & (INT_MAX - 1));
 }
 
+unsigned int VarRefHash_untyped(const void *ref,
+                                unsigned int seed ARG_UNUSED,
+                                unsigned int max)
+{
+    return VarRefHash(ref) & (max - 1);
+}
```

Comment:
Sounds good. Another thing to keep in mind regarding this is performance. Since the bucket number is fixed, the hash map has the potential to be quite slow when you get variable numbers in the millions. Might be worth autogenerating a test which has millions of randomly named variables and comparing the difference in running time. We know that some customers have a ridiculous number of variables.

At the very least take a look at the running time of a host and compare the same host with the running time on master. Solaris 11 is a good candidate for that, since it's only running a single VM ATM, so is relatively disturbance free.